### PR TITLE
added the changes for SyncIQ global settings

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
added a new resource for SyncIQ global settings

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME

synciq_global_settings

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 80% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [X] Acceptance tests

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Comment by AnikaAgiwal2711** [comment]
image

File: powerscale/provider/synciq_global_settings_resource.go [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+
+	if !plan.ReportEmail.IsUnknown() {
+		diags.Append(plan.ReportEmail.ElementsAs(ctx, &toUpdate.ReportEmail, true)...)
+	}
+
+	if toUpdate.SourceNetwork != nil {
+		if (toUpdate.SourceNetwork.Subnet == "" && toUpdate.SourceNetwork.Pool != "") || (toUpdate.SourceNetwork.Subnet != "" && toUpdate.SourceNetwork.Pool == "") {
+			resp.Diagnostics.AddError(
+				"Valid value for both subnet and pool needs to be provided",
+				"Please either provide value for both subnet and pool or else keep both as empty",
```
Comment (Krunal-Thakkar) [inline powerscale/provider/synciq_global_settings_resource.go:306]: Looks like Create and Update logic same so can we write one common function and same for Read import if possible

File: powerscale/provider/synciq_global_settings_resource.go [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+	client *client.Client
+}
+
+// Metadata describes the resource arguments.
+func (r *SyncIQGlobalSettingsResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_synciq_global_settings"
+}
+
+// Schema describes the resource arguments.
+func (r *SyncIQGlobalSettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```
Comment (shenda1) [inline powerscale/provider/synciq_global_settings_resource.go:63]: What is the behaviour if nothing is specified?

File: powerscale/provider/synciq_global_settings_resource.go [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+				},
+			},
+			"tw_chkpt_interval": schema.Int64Attribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "The interval (in seconds) in which treewalk syncs are forced to checkpoint.",
+				MarkdownDescription: "The interval (in seconds) in which treewalk syncs are forced to checkpoint.",
+			},
+			"password": schema.StringAttribute{
+				Optional:            true,
```
Comment (shenda1) [inline powerscale/provider/synciq_global_settings_resource.go:243]: Can we mark it as sensitive?

File: powerscale/provider/synciq_global_settings_resource.go [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+		)
+		return
+	}
+
+	globalSetting, err := helper.GetSyncIQGlobalSettings(ctx, r.client)
+	if err != nil {
+		errStr := constants.ReadSyncIQGlobalSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error updating synciq global settings",
```
Comment (shenda1) [inline powerscale/provider/synciq_global_settings_resource.go:328]: Check the error message

File: powerscale/provider/synciq_global_settings_resource.go (hunk 1 above)
Comment (AnikaAgiwal2711) [reply powerscale/provider/synciq_global_settings_resource.go]: Done

File: powerscale/provider/synciq_global_settings_resource.go (hunk 4 above)
Comment (AnikaAgiwal2711) [reply powerscale/provider/synciq_global_settings_resource.go]: Updated

File: powerscale/provider/synciq_global_settings_resource.go (hunk 2 above)
Comment (AnikaAgiwal2711) [reply powerscale/provider/synciq_global_settings_resource.go:86]: added the changes for that

Comment (AnikaAgiwal2711) [reply powerscale/provider/synciq_global_settings_resource.go:86]: image

**Review by shenda1 — APPROVED** [review]

**Comment by AnikaAgiwal2711** [comment]
image

**Comment by AnikaAgiwal2711** [comment]
run e2e test SyncIQGlobalSettings

**Comment by rounak-adhikary** [comment]
run e2e test SyncIQGlobalSettings

**Pull request merged**